File: trashAdder.js
const fs = require('fs');

fs.readFile('config.json', 'utf8', (err, data) => {
    if (err) {
        console.error('Error reading config file:', err);
        return;
    }

    const configData = JSON.parse(data);
    const filename = configData.filename;
    const trashCharacters = configData.trashCharacters;

    function trashAdder(text) {
        const result = [];
        for (let i = 0; i < text.length; i += 2) {
            result.push(text.slice(i, i + 2));
            if (i + 2 < text.length) {
                result.push(trashCharacters[Math.floor(Math.random() * trashCharacters.length)]);
            }
        }
        return result.join('');
    }

    // Load input text from the file
    fs.readFile(filename, 'utf8', (err, inputText) => {
        if (err) {
            console.error('Error reading input text file:', err);
            return;
        }

        const modifiedText = trashAdder(inputText);
        console.log(modifiedText);
    });

});

File: trashAdder.py
import random, json

with open('config.json', 'r', encoding='utf-8') as config_file:
    config_data = json.load(config_file)
    
    print(config_data)

filename = config_data["filename"]
trashCharacters = config_data["trashCharacters"]

def trashAdder(text):
    result = []
    for i in range(0, len(text), 2):
        result.append(text[i:i+2])
        if i+2 < len(text):
            result.append(random.choice(trashCharacters))
            
    return ''.join(result)


with open(filename, 'r', encoding='utf-8') as text_file:
    input_text = text_file.read()

    modified_text = trashAdder(input_text)
    print(modified_text)
